clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -799,8 +799,11 @@ public void subscribe(Collection<String> topics, ConsumerRebalanceListener liste
                 // treat subscribing to empty topic list as the same as unsubscribing
                 this.unsubscribe();
             } else {
-                log.debug("Subscribed to topic(s): {}", Utils.join(topics, ", "));
-                this.subscriptions.subscribe(topics, listener);
+                Collection<String> nonNullTopics = new ArrayList<>(topics);
```

Comment:
I think it might be better to raise an exception than to silently discard nulls.